### DISCUSSION TITLE
Force variables to appear in expression

### DISCUSSION THREAD

**Discussion by zhuyi-bjut** [MilesCranmer/PySR#572] [General]
Hi！  @MilesCranmer 
When I do the symbolic regression task, I want multiple variables in the target formula, but the results given by pysr are often fewer variables ( 1 ~ 2 ). How to constrain so that there are enough variables I want in the result formula ( 3 ~ 4 ) ? Although they should not appear from the fitting point of view, they have physical meaning from the physical point of view.
Thanks.

**Comment by MilesCranmer** [comment]
See https://github.com/MilesCranmer/PySR/discussions/273#discussioncomment-5231414 for some relevant discussion.

There's also https://github.com/MilesCranmer/PySR/issues/44 but that one is much older and a bit out-of-date.
